manga_translator/translators/chatgpt.py

```diff
@@ -136,22 +136,35 @@ async def _request_translation(self, prompt: str) -> str:
         self.token_count_last = response.usage['total_tokens']
         return response.choices[0].text
 
-
-class GPT35TurboTranslator(GPT3Translator):
+class GPTChatTranslator(GPT3Translator):
     _MAX_REQUESTS_PER_MINUTE = 200
     PROMPT_TEMPLATE = SIMPLE_PROMPT_TEMPLATE
+    _RETRY_ATTEMPTS = 5
+    _model: Literal['gpt-3.5-turbo', 'gpt-4']
 
     async def _request_translation(self, prompt: str) -> str:
         messages = [
             {'role': 'system', 'content': 'You are a professional translator who will follow the required format for translation.'},
             {'role': 'user', 'content': prompt},
         ]
-        response = openai.ChatCompletion.create(
-            model='gpt-3.5-turbo',
-            messages=messages,
-            max_tokens=2048,
-            temperature=self.temperature,
-        )
+
+        # Due to load on OpenAI servers, this fails relatively often, so retry a few times
+        for i in range(self._RETRY_ATTEMPTS):
+            try:
+                response = openai.ChatCompletion.create(
+                    model=self._model,
+                    messages=messages,
+                    max_tokens=2048,
+                    temperature=self.temperature,
+                )
+                break
+            except:
+                if i == self._RETRY_ATTEMPTS - 1:
+                    raise Exception('Failed to get response from OpenAI servers. Use a different translator or try again later.')
+                self.logger.warn(f'Failed to get response from OpenAI servers. Retrying... Attempt: {i+1}/{self._RETRY_ATTEMPTS}')
+                await asyncio.sleep(1)
+                continue
```

Comment:
Retries are being done in the super method already. Theres no need to change anything there but if you want you can add _RETRY_ATTEMPTS.

Reply:
I'd like to keep the retry, mostly due to the fact that during my testing of the GPT-4 translator it failed so often that I thought something was wrong with my code, but I'll reduce it to 3.

Reply:
Instead you can add _RETRY_ATTEMPTS set to 3 to the super class and overwrite it to 5 in the gpt4 class.
Tge reason i prefer the retry implementation in the super class is that it implements a timeout so that requests that take to long will be aborted and rerequested.